src/ui/common/src/components/layouts/menuSidebar.tsx

```diff
@@ -123,22 +123,19 @@ const MenuSidebar: React.FC<{
   const [versionNumber, setVersionNumber] = useState('');
   const location = useLocation();
 
+  const { data } = useEnvironmentGetQuery({ apiKey: user.apiKey } as any, {
+    skip: !user?.apiKey,
+  });
+
   useEffect(() => {
     setCurrentPage(location.pathname);
   }, [dispatch, location.pathname]);
 
   useEffect(() => {
-    async function fetchVersionNumber() {
-      const res = await fetch(`${apiAddress}/api/version`, {
-        method: 'GET',
-        headers: { 'api-key': user.apiKey },
-      });
-      const versionNumberResponse = await res.json();
-      setVersionNumber(versionNumberResponse.version);
+    if (data) {
+      setVersionNumber(data.version);
```

Comment:
will the user or other things set the version number? Otherwise we can just do `const versionNumber = data?.version`

Reply:
Nope, I will remove this useEffect and state variable.
Vestiges here from the infinite loop that I was facing with the Kubernetes dialog :)